lessons/creating-apis-with-python-and-flask.md

```diff
@@ -0,0 +1,713 @@
+
+Our path is:
+
+	/search/pages/results/
+
+If we combine the base URL and the path together into one URL, we'll have created a request to the Chronicling America API that returns all available data in the database:
+
+	http://chroniclingamerica.loc.gov/search/pages/results/
+
+If you [visit the link above](http://chroniclingamerica.loc.gov/search/pages/results/), you'll see all items available in Chronicling America (12,243,633 at the time of writing), , not just the entries related to our search term, "fire." This request also returns a formatted HTML view, rather than the structured view we want to use to collect data.
+
+According to the Chronicling America documentation, in order to get structured data specifically relating to fire, we need to pass one more kind of data in our request: **query parameters**.
+
+	http://chroniclingamerica.loc.gov/search/pages/results/?format=json&proxtext=fire
+
+The query parameters follow the `?` in the request, and are seperated from one another by the `&` symbol. The first query parameter, `format=json`, changes the returned data from HTML to JSON. The second, `proxtext=fire`, narrows the returned entries to those that include our search term.
+
+If you [follow the above link](http://chroniclingamerica.loc.gov/search/pages/results/?format=json&proxtext=fire) in your browser, you'll see a structured list of the items in the database related to the search term "fire." The format of the returned data is called JSON, and is a structured format that looks like this excerpt from the Chronicling America results:
+
+```json
+"city": [
+        "Washington"
+      ],
+      "date": "19220730",
+      "title": "The Washington Herald.",
+      "end_year": 1939,
+```
+
+By making requests to the Chronicling America API, we've accessed information on news stories that contain the search term "fire," and returned data that includes the date of publication and the page the article appears on. If we were to pursue this research question further, a next step might be finding how many stories relating to fire appear on a newspaper's front page over time, or perhaps cleaning the data to reduce the number of false positives. As we have seen, however, exploring an API can be a useful first step in gathering data to tackle a research question.
+
+<div class="alert alert-warning">
+Note that in this section, we skipped an important step: finding an appropriate API in the first place. Some resources for researching APIs are available at the end of this lesson.
+</div>
+
+## What Users Want in an API
+
+As we've learned, documentation is a user's starting place when working with a new API, and well-designed URLs make it easier for users to intuitively find resources. Because they help users to quickly access information through your API, these elements—documentation and well-conceived URLs—are the *sine qua non* of a good API. We'll discuss these elements in greater depth later in this tutorial.
```

Comment:
*sine qua non* - I'm a ignoramus who sometimes stumbles over things like this. Could we just say "The essential element"?

Reply:
Deferring to Patrick.